pandas/tests/arrays/integer/test_construction.py

```diff
@@ -4,8 +4,8 @@
 import pandas as pd
 import pandas._testing as tm
 from pandas.api.types import is_integer
-from pandas.core.arrays import IntegerArray, integer_array
-from pandas.core.arrays.integer import Int8Dtype, Int32Dtype, Int64Dtype
+from pandas.core.arrays import IntegerArray
+from pandas.core.arrays.integer import Int8Dtype, Int32Dtype, Int64Dtype, integer_array
```

Comment:
shouldn't this be removed?

Reply:
> shouldn't this be removed?

see the top post, this is the only file were I for now kept `integer_array` (it's used a lot, and it is specifically testing the behaviour of the construction implemented by it). But just changed the import, because I removed it from `pandas.core.arrays` 
Now, as said above, I can further remove it in this file as well (can do that here, or later in another PR)

Reply:
i would completely remove it in this PR

Reply:
OK, I removed any remaining usage of `integer_array` in this file, so could also remove the actual function.

I replaced it with a mixture of `pd.array` and `IntegerArray._from_sequence` (and sometimes a parametrized fixture to test both), since they are not exactly equal. For example, with `_from_sequence` you can also pass it a float array without specifying a dtype, something you can't replicate by only testing `pd.array(..)`